api/httpjson/RPCserver.go

```diff
@@ -47,22 +41,26 @@ type ServeMux struct {
 
 // NewServer will create a new RPC server instance.
 func NewServer(localNode *node.LocalNode, wallet vault.Wallet) *RPCServer {
-	server := &RPCServer{
+	return &RPCServer{
 		mainMux: ServeMux{
 			m: make(map[string]common.Handler),
 		},
 		httpListener:  ":" + strconv.Itoa(int(config.Parameters.HttpJsonPort)),
 		httpsListener: ":" + strconv.Itoa(int(config.Parameters.HttpsJsonPort)),
 		localNode:     localNode,
 		wallet:        wallet,
+		limiter:       rate.NewLimiter(rate.Limit(config.Parameters.RPCRateLimit), int(config.Parameters.RPCRateBurst)),
 	}
-
-	return server
 }
 
-// this is the funciton that should be called in order to answer an rpc call
+// Handle is the funciton that should be called in order to answer an rpc call
 // should be registered like "http.HandleFunc("/", httpjsonrpc.Handle)"
 func (s *RPCServer) Handle(w http.ResponseWriter, r *http.Request) {
+	if !s.limiter.Allow() {
+		w.WriteHeader(http.StatusTooManyRequests)
```

Comment:
Will this affect legal node's connection when illegal node exists?

Reply:
It might be if node is under DDoS attack, but a node cannot distinguish between honest node and attacker node, so he can only reject all of them